### PR TITLE
Fix for teapot slide issue #46

### PR DESCRIPTION
**Pull request by deepakrawat22**
This pull request fixes the teapot  slide issue. Following changes made this possible:

1 - Updated the sample files from vuforia to latest.
2 - Custom shader for AR objects as they use MVP matrix obtained from vuforia task.
3 - Corrected the camera pass through object dimensions. The size ratio of GL viewport and camera passthrough must match.

GearVRf-DCO-1.0-Signed-off-by: Deepak Rawat
deepak.rawat@samsung.com

### CODE REVIEW

**Comment by liaxim** [comment]
What is the exact fix for the slide?

--
I think having the camera passthrough as a full screen quad yields a very distorted image while wearing the headset. I still notice sliding going on but it is harder to tell. What was wrong with the old passthrough dimensions? The teapot does not move smoothly anymore, it is kind of stuttering.

**Comment by deepakrawat22** [comment]
The fix for the slide is to pass the MVP matrix obtained from vuforia, so that's why the custom shader. Another thing that's need to be done is to make the GL viewport match the one obtained from vuforia which depends on camera passthrough.

**Comment by liaxim** [comment]
@deepakrawat22 Do you think we should try it out and review again?

**Comment by deepakrawat22** [comment]
@liaxim I am still doing some changes, will commit them in some time and let you know once its ready for review. Thanks.

**Comment by deepakrawat22** [comment]
@liaxim this pull request is ready for review now. Need the PR #630 from GearVRf to test the new commit. Thanks.

File: gvr-vuforia/app/src/main/java/org/gearvrf/vuforiasample/VuforiaSampleMain.java [hunk 1]
```diff
@@ -169,7 +187,12 @@ private void hideTeapot() {
         }
     }
 
-    public void updateObjectPose(State state) {
+    public void updateObjectPose() {
+        State state = mRenderer.begin();
+
+        // viewport as per default 1024 x 1024 window size
+        mainScene.getMainCameraRig().setViewport(0, 224, 1024, 576);
```
Comment (liaxim) [inline gvr-vuforia/app/src/main/java/org/gearvrf/vuforiasample/VuforiaSampleMain.java]: How did you determine these numbers?

**Comment by liaxim** [comment]
This is much better.

@teresa-mao If you have time please check this out.

@thomasflynn You might want to see how this works now.

File: gvr-vuforia/app/src/main/java/org/gearvrf/vuforiasample/VuforiaSampleMain.java (hunk 1 above)
Comment (deepakrawat22) [inline gvr-vuforia/app/src/main/java/org/gearvrf/vuforiasample/VuforiaSampleMain.java]: The camera resolution is 2560 X 1440 giving aspect ratio as 16:9. The GL viewport aspect ratio also match. So just change 1024 X 1024 to 16:9

**Comment by liaxim** [comment]
Teresa noticed that the aspect ratio of the camera view is really off in monoscopic mode.

**Comment by deepakrawat22** [comment]
Finally corrected the code and now generate texture co-ordinates based on device aspect ration. Now its working on both S7 and Note4. Still need GearVRf #630 to work.

**Comment by deepakrawat22** [comment]
This pull request is not yet ready for review as need to change the setting the viewport technique.

**Comment by deepakrawat22** [comment]
@liaxim Removed the viewport set call as viewport is set from config file.

**Comment by liaxim** [comment]
@teresa-mao Please let us know how does it work for you. Thanks.

**Comment by liaxim** [comment]
Looks good. The teapot could move a bit smoother. On Note5 the teapot is flipping oddly occasionally; not observed on S7. We can continue improving as the time permits.

**Comment by liaxim** [comment]
Will merge in the afternoon if there are no objections.

**Comment by teresa-mao** [comment]
yeah we should merge it & continue to make improvements later.

**Pull request merged**